vllm/v1/engine/async_llm.py

```diff
@@ -345,9 +347,12 @@ async def _run_output_handler(self):
 
     async def abort(self, request_id: str) -> None:
         """Abort RequestId in OutputProcessor and EngineCore."""
-
-        request_ids = self.output_processor.abort_requests((request_id, ))
+        request_ids = self.output_processor.flatten_req_to_abort(
+            (request_id, ))
         await self.engine_core.abort_requests_async(request_ids)
+        # At this point, the abort message has already been sent to EngineCore,
```

Comment:
Can you update this comment to explain why this ordering is important for the race condition?

Reply:
done